File: frontend/src/lib/api/notifications.ts
import { apiClient } from "@/lib/api/client";

export type NotificationKind =
  | "new_booking"
  | "whatsapp_sent"
  | "review_waiting"
  | "escalation";

export type NotificationItem = {
  id: string;
  kind: NotificationKind;
  title: string;
  detail?: string;
  created_at: string;
  read: boolean;
  href?: string;
};

export type NotificationsResponse = {
  items: NotificationItem[];
  unread: number;
};

export async function fetchNotifications(): Promise<NotificationsResponse> {
  const { data } = await apiClient.get<NotificationsResponse>("/api/notifications");
  return data;
}

export async function markNotificationsRead(): Promise<void> {
  await apiClient.post("/api/notifications/read-all");
}

File: backend/api/schemas/notifications.py
"""DTOs für das Benachrichtigungs-Panel (Glocke)."""

from __future__ import annotations

from pydantic import BaseModel, Field


class NotificationItem(BaseModel):
    """Ein Eintrag im Glocken-Panel."""

    id: str
    kind: str  # new_booking | whatsapp_sent | review_waiting | escalation
    title: str
    detail: str = ""
    created_at: str
    read: bool = False
    href: str | None = None


class NotificationsResponse(BaseModel):
    """Feed + Anzahl ungelesener Einträge."""

    items: list[NotificationItem] = Field(default_factory=list)
    unread: int = 0
